### ISSUE TITLE
Is there a possibility to export types of the library into the single file?

### ISSUE THREAD

**Issue by kazeens** [Automattic/mongoose#13268]
### Prerequisites

- [X] I have written a descriptive issue title

### Mongoose version

7.0.0 

### Node.js version

19

### MongoDB version

5.0

### Operating system

None

### Issue

Hello.

I am developing deno application.
Since deno started supporting npm modules and mongoose has become availiable in deno ecosystem which is great, the typing, in particular, are not available(for any npm modules). 

So what I am trying to do is to manually export types of mongoose  library and use its  declaration file in my project.

Already tried utility `npm-dts` which is great and extracted d.ts files of some libraries, but with mongoose repo it fails to do so.

So basically I am trying to find solution for mongoose types in deno.
Maybe there is a better way?
Maybe I just need to use some different method/tool.

Would be really grateful to get any solid response on that issue.
Thank you

**Comment by vkarpov15** [comment]
Can you please explain a little more what you're trying to achieve? We do have typings in this project in the [types directory](https://github.com/Automattic/mongoose/tree/master/types), does this not work in deno for some reason?
